vars/buildAndProvideDebianPackage.groovy

```diff
@@ -0,0 +1,69 @@
+String stashFileList = '*.gz,*.bz2,*.xz,*.deb,*.ddeb,*.udeb,*.dsc,*.changes,*.buildinfo,lintian.txt'
+String archiveFileList = '*.gz,*.bz2,*.xz,*.deb,*.ddeb,*.udeb,*.dsc,*.changes,*.buildinfo'
+
+def call(Boolean isArchIndependent = false) {
+  pipeline {
+    agent any
+    stages {
+      stage('Build source') {
+        steps {
+          sh '/usr/bin/build-source.sh'
+          stash(name: 'source', includes: stashFileList)
+          cleanWs(cleanWhenAborted: true, cleanWhenFailure: true, cleanWhenNotBuilt: true, cleanWhenSuccess: true, cleanWhenUnstable: true, deleteDirs: true)
+        }
+      }
+      stage('Build binary') {
+        parallel {
+          stage('Build binary - armhf') {
+            agent { label 'arm64' }
+            when { expression { return !isArchIndependent } }
+            steps {
+              cleanWs(cleanWhenAborted: true, cleanWhenFailure: true, cleanWhenNotBuilt: true, cleanWhenSuccess: true, cleanWhenUnstable: true, deleteDirs: true)
+              unstash 'source'
+              sh 'architecture="armhf" build-binary.sh'
+              stash(includes: stashFileList, name: 'build-armhf')
+              cleanWs(cleanWhenAborted: true, cleanWhenFailure: true, cleanWhenNotBuilt: true, cleanWhenSuccess: true, cleanWhenUnstable: true, deleteDirs: true)
+            }
+          }
+          stage('Build binary - arm64') {
+            agent { label 'arm64' }
+            // Always run; arch-independent packages are built here.
```

Comment:
We might want to do arch independent packages on amd64 as its much faster

We can achieve this by setting
```
export architecture="all"
export FORCE_BINARY_ONLY="all"
```

Reply:
Maybe it's better to set `MAIN_ARCHITECTURE` correctly in `/etc/jenkins/debian_glue`? This way, packages with a mix of arch-dependent & arch-independent binary packages will benefit, too.

The condition & missing of it has to match the architecture that will build arch-independent packages.

Reply:
This as been done, "all" is now built on amd64